compiler+runtime/src/cpp/jank/read/parse.cpp

```diff
@@ -15,6 +15,81 @@ namespace jank::read::parse
 {
   using namespace jank::runtime;
 
+  result<option<native_persistent_string>, native_persistent_string>
```

Comment:
`result<option<T>, E>` implies that `ok(none)` is expected, but we're never returning that. Why do we have the `option<T>` in there at all?